src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs

```diff
@@ -16,9 +17,11 @@ internal class EnumConverter<T> : JsonConverter<T>
         // Odd type codes are conveniently signed types (for enum backing types).
         private static readonly string? s_negativeSign = ((int)s_enumTypeCode % 2) == 0 ? null : NumberFormatInfo.CurrentInfo.NegativeSign;
 
+        private const string ValueSeparator = ", ";
+
         private readonly EnumConverterOptions _converterOptions;
-        private readonly JsonNamingPolicy _namingPolicy;
-        private readonly ConcurrentDictionary<string, string>? _nameCache;
+        private readonly JsonNamingPolicy? _namingPolicy;
+        private readonly ConcurrentDictionary<T, JsonEncodedText> _nameCache = new ConcurrentDictionary<T, JsonEncodedText>();
```

Comment:
What does `T` add (instead of string?). Seems like there could be extra JITting for every `T` now (i.e. for every Enum type).

Reply:
Per offline discussion, I ran a simple console app test measuring private bytes and peak private bytes using 4 dictionaries -- one set that had a different Enum type for the key, and another set using ulong for the key. Ran against crossgen'd \ ready-to-run assemblies.

| private | peak private | description |
| --- | --- | --- |
| 3900 | 6732 | 4 dictionaries each with ulong as key
| 3948 | 6792 | 4 dictionaries each with a different enum type as key
```

So 16K+- extra private bytes per dictionary type which I assume is memory to hold JITted bytes -- the physical size of the assemblies did not get larger, so crossgen didn't expand\duplicate the dictionary code (I expected it would however)

Reply:
I've updated to use `ulong` as the key.